File: crates/harn-hostlib/tests/fixtures/ast/typescript/source.ts
interface Greeter {
    greet(name: string): string;
}

class HelloGreeter implements Greeter {
    greet(name: string): string {
        return `hello ${name}`;
    }
}

function shout(message: string): string {
    return message.toUpperCase();
}

const yell = (message: string) => shout(message);

type Maybe<T> = T | null;

File: crates/harn-hostlib/tests/fixtures/ast/rust/source.rs
struct Greeter {
    name: String,
}

enum Color {
    Red,
    Green,
    Blue,
}

trait Speak {
    fn speak(&self) -> String;
}

impl Greeter {
    fn greet(&self) -> String {
        format!("hello {}", self.name)
    }
}

fn shout(s: &str) -> String {
    s.to_uppercase()
}

type Counter = i64;
